File: modules/print_metadata.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

class Program {
    static async Task Main(string[] args) {
        var client = new HttpClient();
        string metadata;

        // check if running on AWS
        try {
            metadata = await client.GetStringAsync("http://169.254.169.254/latest/meta-data/");
            Console.WriteLine("Running on AWS, metadata: " + metadata);
        } catch (HttpRequestException) {
            // not running on AWS
        }

        // check if running on Azure
        try {
            metadata = await client.GetStringAsync("http://169.254.169.254/metadata/instance?api-version=2020-09-01");
            Console.WriteLine("Running on Azure, metadata: " + metadata);
        } catch (HttpRequestException) {
            // not running on Azure
        }

        // check if running on GCP
        try {
            metadata = await client.GetStringAsync("http://metadata.google.internal/computeMetadata/v1/");
            Console.WriteLine("Running on GCP, metadata: " + metadata);
        } catch (HttpRequestException) {
            // not running on GCP
        }
    }
}

File: modules/print_metadata.py
import requests

def get_metadata(url):
    try:
        response = requests.get(url, timeout=1)
        if response.status_code == 200:
            return response.text
    except requests.exceptions.RequestException as e:
        pass

aws_metadata = get_metadata("http://169.254.169.254/latest/meta-data/")
if aws_metadata:
    print("Running on AWS, metadata: " + aws_metadata)
else:
    azure_metadata = get_metadata("http://169.254.169.254/metadata/instance?api-version=2020-09-01")
    if azure_metadata:
        print("Running on Azure, metadata: " + azure_metadata)
    else:
        gcp_metadata = get_metadata("http://metadata.google.internal/computeMetadata/v1/")
        if gcp_metadata:
            print("Running on GCP, metadata: " + gcp_metadata)
